internal/cqarrow/json.go

```diff
@@ -0,0 +1,173 @@
+package cqarrow
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/goccy/go-json"
+
+	"github.com/apache/arrow/go/v12/arrow"
+	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/apache/arrow/go/v12/arrow/memory"
+)
+
+type JSONBuilder struct {
+	*array.ExtensionBuilder
+	dtype *JSONType
+}
+
+func NewJSONBuilder(mem memory.Allocator, dtype arrow.ExtensionType) *JSONBuilder {
+	b := &JSONBuilder{
+		ExtensionBuilder: array.NewExtensionBuilder(mem, dtype),
+		dtype:            dtype.(*JSONType),
+	}
+	return b
+}
+
+func (b *JSONBuilder) Append(v any) {
+	if v == nil {
+		b.AppendNull()
+		return
+	}
+	bytes, err := json.Marshal(v)
```

Comment:
I'm wondering if we need to use json.Encoder here as well, like we had to in plugin-sdk (https://github.com/cloudquery/plugin-sdk/pull/714/files)?